### PR TITLE
[Rule Tuning] Admin Role Assigned to User - Google Workspace

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/2226

## Summary
This rule needs tuning to identify when an admin role is delegated or assigned to a user in Google Workspace. The original rule only identifies when a Google Workspace administrator assigns a role to a user, not a specific admin role. This rule update will identify if any admin role (mobile, app, console, etc.) is assigned to a user.

Example Document:

<details>

```
{
  "_index": ".ds-logs-google_workspace.admin-default-2022.07.25-000001",
  "_id": "uQ5y9OnfK7QBeAqXVdORwaQsdlw=",
  "_version": 1,
  "_score": 0,
  "_source": {
    "agent": {
      "name": "ubuntu-server-tdejesus",
      "id": "f4011165-a1b8-4d8e-b902-56ba8835cc28",
      "ephemeral_id": "3d0af180-a1ca-4a5f-9153-2e986dc98e25",
      "type": "filebeat",
      "version": "8.2.3"
    },
    "elastic_agent": {
      "id": "f4011165-a1b8-4d8e-b902-56ba8835cc28",
      "version": "8.2.3",
      "snapshot": false
    },
    "source": {
      "geo": {
        "continent_name": "North America",
        "region_iso_code": "US-OH",
        "city_name": "North Canton",
        "country_iso_code": "US",
        "country_name": "United States",
        "region_name": "Ohio",
        "location": {
          "lon": -81.3798,
          "lat": 40.7961
        }
      },
      "as": {
        "number": 10796,
        "organization": {
          "name": "TWC-10796-MIDWEST"
        }
      },
      "ip": "2603:6010:ed07:5cb7:6d05:186b:2925:f1d1",
      "user": {
        "domain": "dejesus-gw-rule-dev.com",
        "name": "contact",
        "id": "115341805336974604065",
        "email": "contact@dejesus-gw-rule-dev.com"
      }
    },
    "tags": [
      "forwarded",
      "google-workspace-admin"
    ],
    "cloud": {
      "availability_zone": "us-east1-b",
      "instance": {
        "name": "ubuntu-server-tdejesus",
        "id": "1709224677170316971"
      },
      "provider": "gcp",
      "service": {
        "name": "GCE"
      },
      "machine": {
        "type": "e2-medium"
      },
      "project": {
        "id": "elastic-security-dev"
      },
      "account": {
        "id": "elastic-security-dev"
      }
    },
    "input": {
      "type": "httpjson"
    },
    "@timestamp": "2022-08-01T16:20:39.177Z",
    "ecs": {
      "version": "8.2.0"
    },
    "related": {
      "ip": [
        "2603:6010:ed07:5cb7:6d05:186b:2925:f1d1"
      ],
      "user": [
        "contact",
        "suspicious"
      ]
    },
    "google_workspace": {
      "actor": {
        "type": "USER"
      },
      "kind": "admin#reports#activity",
      "admin": {
        "role": {
          "name": "_MOBILE_ADMIN_ROLE"
        },
        "user": {
          "email": "suspicious@dejesus-gw-rule-dev.com"
        }
      },
      "event": {
        "type": "DELEGATED_ADMIN_SETTINGS"
      }
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "google_workspace.admin"
    },
    "organization": {
      "id": "C02aasz56"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-08-01T16:26:16Z",
      "provider": "admin",
      "created": "2022-08-01T16:26:15.190Z",
      "action": "ASSIGN_ROLE",
      "id": "-7067121314811828393",
      "category": [
        "iam"
      ],
      "type": [
        "change"
      ],
      "dataset": "google_workspace.admin"
    },
    "user": {
      "domain": "dejesus-gw-rule-dev.com",
      "name": "contact",
      "id": "115341805336974604065",
      "target": {
        "domain": "dejesus-gw-rule-dev.com",
        "name": "suspicious",
        "email": "suspicious@dejesus-gw-rule-dev.com"
      }
    }
  },
  "fields": {
    "event.category": [
      "iam"
    ],
    "elastic_agent.version": [
      "8.2.3"
    ],
    "source.user.email": [
      "contact@dejesus-gw-rule-dev.com"
    ],
    "cloud.availability_zone": [
      "us-east1-b"
    ],
    "source.user.name.text": [
      "contact"
    ],
    "user.target.email": [
      "suspicious@dejesus-gw-rule-dev.com"
    ],
    "source.geo.region_name": [
      "Ohio"
    ],
    "google_workspace.actor.type": [
      "USER"
    ],
    "source.ip": [
      "2603:6010:ed07:5cb7:6d05:186b:2925:f1d1"
    ],
    "agent.name": [
      "ubuntu-server-tdejesus"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "source.geo.region_iso_code": [
      "US-OH"
    ],
    "user.target.name.text": [
      "suspicious"
    ],
    "google_workspace.event.type": [
      "DELEGATED_ADMIN_SETTINGS"
    ],
    "source.geo.city_name": [
      "North Canton"
    ],
    "user.id": [
      "115341805336974604065"
    ],
    "input.type": [
      "httpjson"
    ],
    "data_stream.type": [
      "logs"
    ],
    "tags": [
      "forwarded",
      "google-workspace-admin"
    ],
    "user.target.name": [
      "suspicious"
    ],
    "related.user": [
      "contact",
      "suspicious"
    ],
    "event.provider": [
      "admin"
    ],
    "cloud.provider": [
      "gcp"
    ],
    "cloud.machine.type": [
      "e2-medium"
    ],
    "cloud.service.name": [
      "GCE"
    ],
    "agent.id": [
      "f4011165-a1b8-4d8e-b902-56ba8835cc28"
    ],
    "ecs.version": [
      "8.2.0"
    ],
    "event.created": [
      "2022-08-01T16:26:15.190Z"
    ],
    "organization.id": [
      "C02aasz56"
    ],
    "google_workspace.admin.user.email": [
      "suspicious@dejesus-gw-rule-dev.com"
    ],
    "agent.version": [
      "8.2.3"
    ],
    "source.user.name": [
      "contact"
    ],
    "source.as.number": [
      10796
    ],
    "user.name": [
      "contact"
    ],
    "source.geo.location": [
      {
        "coordinates": [
          -81.3798,
          40.7961
        ],
        "type": "Point"
      }
    ],
    "cloud.instance.id": [
      "1709224677170316971"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "google_workspace"
    ],
    "related.ip": [
      "2603:6010:ed07:5cb7:6d05:186b:2925:f1d1"
    ],
    "source.geo.country_iso_code": [
      "US"
    ],
    "user.target.domain": [
      "dejesus-gw-rule-dev.com"
    ],
    "source.user.id": [
      "115341805336974604065"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "user.domain": [
      "dejesus-gw-rule-dev.com"
    ],
    "source.as.organization.name.text": [
      "TWC-10796-MIDWEST"
    ],
    "elastic_agent.id": [
      "f4011165-a1b8-4d8e-b902-56ba8835cc28"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "source.as.organization.name": [
      "TWC-10796-MIDWEST"
    ],
    "source.geo.continent_name": [
      "North America"
    ],
    "google_workspace.kind": [
      "admin#reports#activity"
    ],
    "event.ingested": [
      "2022-08-01T16:26:16.000Z"
    ],
    "event.action": [
      "ASSIGN_ROLE"
    ],
    "@timestamp": [
      "2022-08-01T16:20:39.177Z"
    ],
    "google_workspace.admin.role.name": [
      "_MOBILE_ADMIN_ROLE"
    ],
    "cloud.account.id": [
      "elastic-security-dev"
    ],
    "event.type": [
      "change"
    ],
    "data_stream.dataset": [
      "google_workspace.admin"
    ],
    "agent.ephemeral_id": [
      "3d0af180-a1ca-4a5f-9153-2e986dc98e25"
    ],
    "source.user.domain": [
      "dejesus-gw-rule-dev.com"
    ],
    "event.id": [
      "-7067121314811828393"
    ],
    "source.geo.country_name": [
      "United States"
    ],
    "event.dataset": [
      "google_workspace.admin"
    ],
    "cloud.instance.name": [
      "ubuntu-server-tdejesus"
    ],
    "cloud.project.id": [
      "elastic-security-dev"
    ],
    "user.name.text": [
      "contact"
    ]
  }
}
```

</details>

Screen Shot 2022-08-25 at 9 52 55 AM

### CODE REVIEW

**Review by imays11 — APPROVED** [review]

File: rules/integrations/google_workspace/persistence_google_workspace_admin_role_assigned_to_user.toml [hunk 1]
```diff
@@ -40,12 +42,21 @@ references = ["https://support.google.com/a/answer/172176?hl=en"]
...
+    "Identity and Access",
+    "Persistence",
+]
 timestamp_override = "event.ingested"
 type = "query"
 
 query = '''
-event.dataset:google_workspace.admin and event.provider:admin and event.category:iam and event.action:ASSIGN_ROLE
+event.dataset:"google_workspace.admin" and event.action:"ASSIGN_ROLE" and event.category:"iam"
+  and google_workspace.event.type:"DELEGATED_ADMIN_SETTINGS" and google_workspace.admin.role.name : *_ADMIN_ROLE
```
Comment (w0rk3r) [inline rules/integrations/google_workspace/persistence_google_workspace_admin_role_assigned_to_user.toml:59]:
```suggestion
event.dataset:"google_workspace.admin" and event.category:"iam" and event.action:"ASSIGN_ROLE"
  and google_workspace.event.type:"DELEGATED_ADMIN_SETTINGS" and google_workspace.admin.role.name : *_ADMIN_ROLE
```

**Review by w0rk3r — APPROVED** [review]
Nice tuning 🔥 , we could consider a severity bump as this should not be a constant behavior

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/integrations/google_workspace/persistence_google_workspace_admin_role_assigned_to_user.toml (hunk 1 above)
Comment (terrancedejesus) [reply rules/integrations/google_workspace/persistence_google_workspace_admin_role_assigned_to_user.toml:59]: Sounds fine to me!

**Pull request merged**